### ISSUE TITLE
bug: gql.operation.name needs to identify operation name properly

### ISSUE THREAD

**Issue by kutysam** [Netflix/dgs-framework#459]
Let's look at a simple graphql query
```
query simpleschemacheck{
  __schema {
    types {
      name
    }
  }
}
```
Summarizing these 2 websites
Web1: https://graphql.org/learn/queries/#operation-name
Web2: https://graphql.org/learn/serving-over-http/#post-request
- `operationName` key within the HTTP POST query is **optional** for single queries and mandatory for multiple queries.
- in the above example, `simpleschemacheck` can be deemed as the operation name as it is a single query. We need not state this in the operationName POST Query.

## Expected behavior
`gql.operation.name` metric is anoynomous

## Actual behavior
`gql.operation.name` metric should be `simplequerycheck` as there is only 1 query.

## Steps to reproduce
- Just use my query and send it via postman.
- graphiql sends operationName even though it has 1 query. It follows the `simpleschemacheck` operation name. However, if there is no `simplequerycheck`, it will not send a `operationName` within the POST request. In this case, anonynomous is definetly correct. (Use chrome debugger and view the network requests)
- Other libraries may not send this operationName within the POST as it is an optional field even though `simpleschemacheck` is the operation name.

## What should be done
- `gql.operation.name` should do the following check.

Check if operationName is present.
```
If operationName is present - gql.operation.name will be operationName
If operationName is not present, this has to be a 1 query post request.
   Check the single query's operation name
      If query's operation name is nonempty, `gql.operation.name` is this operation name
      ElseIf the query operation name is empty, `gql.operation.name` will be anonynomous.
```
In other words, there has to be a proper mapping with operationName and query <operation_name>

## Why do we need this?
- Libraries are sending healthchecks without operationName while having a `query operationname`.
- DGS metrics is marking `gql.operation.name` as `anoynomous`. It makes metrics very hard to read as we do not know the actual operation name that is in the query and thus, we can't filter out healthchecks and normal requests.

**Comment by berngp** [comment]
Thanks for the detailed explanation.

**Comment by berngp** [comment]
Sorry for the delay, PR #491 open.

**Comment by kutysam** [comment]
Hi @berngp ,
We have tested it and I'm glad to say its working very nicely now. We are able to see the operationname within the metrics for those that are not 'anonynomous'
Really appreciate the effort and fixing this up 💪  Thanks!
